src/ServiceControl/DbMigrations/1.27/SplitFailedMessageDocumentsMigration.cs

```diff
@@ -0,0 +1,158 @@
+﻿namespace Particular.ServiceControl.DbMigrations
+{
+    using System.Linq;
+    using global::ServiceControl;
+    using global::ServiceControl.MessageFailures;
+    using global::ServiceControl.Operations;
+    using global::ServiceControl.Recoverability;
+    using NServiceBus.ObjectBuilder;
+    using Raven.Client;
+
+    public class SplitFailedMessageDocumentsMigration : IMigration
+    {
+        public SplitFailedMessageDocumentsMigration(IBuilder builder)
+        {
+            var failedEnrichers = builder.BuildAll<IFailedMessageEnricher>().ToArray();
+
+            failedMessageFactory = new FailedMessageFactory(failedEnrichers);
+        }
+
+        public string Apply(IDocumentStore store)
+        {
+            var currentPage = 0;
+            int retrievedResults;
+
+            var stats = new MigrationStats();
+
+            do
+            {
+                using (var session = store.OpenSession())
+                {
+
+                    var failedMessages = session.Advanced.LoadStartingWith<FailedMessage>(
```

Comment:
@tmasternak As soon as you add an orderby clause you aren't doing a load any more, you are doing a query. Doing a query in Raven requires that we add an index. I believe that `LoadStartingWith` should give us all of the documents in a deterministic order (insertion order?) but I can't find any doco proving that.